src/redux/reducers/favoriteDrinkReducer.js

```diff
@@ -6,9 +6,14 @@ const initialState = {
 export const favoriteDrinksReducer = (state = initialState, action) => {
     switch (action.type) {
         case 'ADD_FAVORITE_DRINK':
-            return {
-                ...state,
-                favoriteDrinks: [...state.favoriteDrinks, action.payload]
+            const foundDrink = state.favoriteDrinks.find(drink => drink.idDrink === action.payload.idDrink)
```

Comment:
glad that this worked! bye bye duplicate favorites!

Reply:
^^ what Colgan said.